File: codes/python/leetcodes/src/main/python/com/hit/basinfo/search_algorithm/OrderSearch.py
"""
 author: Charies Gavin
 date: 11/8/19,3:46 PM
 https:github.com/guobinhit
 description: Order Search
"""


def order_search(nums, target):
    if len(nums) == 0:
        return -1
    elif len(nums) < 2 and nums[0] != target:
        return -1

    index = 0
    while index < len(nums):
        if nums[index] == target:
            return index
        index += 1

    return -1


nums = [-1, 0, 1, 2, 3]
target = -1
print order_search(nums, target)

File: codes/java/leetcodes/src/main/java/com/hit/basinfo/search_algorithm/OrderSearch.java
package com.hit.basinfo.search_algorithm;

/**
 * author:Charies Gavin
 * date:2019/1/7,15:00
 * https:github.com/guobinhit
 * description: Order Search
 */
public class OrderSearch {
    /**
     * Order Search
     *
     * @param nums   pending query array
     * @param target target element
     * @return target element index
     */
    public int orderSearch(int[] nums, int target) {
        // check parameters
        if (nums == null) {
            return -1;
        }
        if (nums.length < 2 && nums[0] != target) {
            return -1;
        }

        // declare variables
        int i = 0;
        /**
         * core algorithm
         * actually only compare every element in array with order to target number
         */
        for (; i < nums.length - 1; i++) {
            if (nums[i] == target) {
                return i;
            }
        }
        return -1;
    }
}
